lib/src/clojure_lsp/kondo.clj

```diff
@@ -114,13 +95,40 @@
        name-end-row
        name-end-col))
 
-(defn normalize-analysis [external? analysis]
+(defn ^:private normalize-analysis [external? analysis]
   (for [[bucket vs] analysis
         v vs
         element (entry->normalized-entries (assoc v :bucket bucket :external? external?))
         :when (valid-element? element)]
     element))
 
+(defn ^:private normalize
```

Comment:
not related with your refactor, but this PR made me realize how much we iterate and "normalize" analysis all the time, I wonder how much time we lose here, maybe there are quick wins

Reply:
Perhaps... something to research sometime.